github/release_notes/model.py

```diff
@@ -83,17 +85,29 @@ def __init__(
         reference_type: ReferenceType,
         reference_id: str,
         user_login: str,
-        source_repo: str,
-        cn_current_repo: ComponentName,
+        current_component: gci.componentmodel.Component,
+        source_component: gci.componentmodel.Component,
     ):
         if reference_id:
-            check_type(reference_id, str)
+            ci.util.check_type(reference_id, str)
+
+        ci.util.not_none(source_component)
 
         reference = Reference(type=reference_type, identifier=reference_id)
 
-        cn_source_repo = ComponentName(name=source_repo)
-        is_current_repo = cn_current_repo == cn_source_repo
-        from_same_github_instance = cn_current_repo.github_host() == cn_source_repo.github_host()
+        source_component_access = cnudie.util.determine_main_source_for_component(
+            source_component
+        ).access
+        current_component_access = cnudie.util.determine_main_source_for_component(
+            current_component
+        ).access
+
+        is_current_repo = current_component.name == source_component.name
```

Comment:
nit: please talk about _components_ (I know, there is currently a 1:1-relationship between components and github-repositories. however this is not necessarily so in all cases in the future, thanks to CNUDIE's more generic model.

Reply:
also, I would prefer to not persist a value that can be trivially calculated. rather calculate it where required (maybe add a method if it is required multiple times)